services/content_generator.py

```
@@ -418,94 +284,211 @@ def _update_title_in_markdown(self, markdown: str, new_title: str) -> str:
+        # Replace event placeholders with AI-generated event descriptions
+        def replace_event(match):
+            event_url = match.group(1)
+            # Generate AI description for the event
+            event_description = self._generate_event_description(event_url, ai_service, force_ai)
+            return event_description
+        
+        markdown = re.sub(r'\[event: ([^\]]+)\]', replace_event, markdown)
+        
+        return markdown
+    
+    def _generate_video_description(self, video_path: str, ai_service, force_ai: bool = False) -> str:
```

Ruff (0.12.2)

Unused method argument: `ai_service`

(ARG002)